### PR TITLE
Add netcfg_diff lookup plugin

### PR DESCRIPTION
**Pull request by ganeshrn**
Fixes https://github.com/ansible-network/network-engine/issues/86
*  Add config diff lookup plugin
*  Add integration test for netcfg_diff
*  Doc update

### CODE REVIEW

**Comment by trishnaguha** [comment]
recheck

File: lookup_plugins/config_diff.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+lookup: config_diff
+author: Ansible Network
+version_added: "2.6"
```
Comment (trishnaguha) [inline lookup_plugins/config_diff.py]: shouldn't this be 2.5?

**Review by trishnaguha — CHANGES_REQUESTED** [review]
please remove the retry file from tests/

File: lookup_plugins/config_diff.py (hunk 1 above)
Comment (ganeshrn) [reply lookup_plugins/config_diff.py]: fixed

**Comment by ganeshrn** [comment]
@trishnaguha Fixed review comments

**Review by trishnaguha — APPROVED** [review]

**Pull request merged**